### DISCUSSION TITLE
X.509 certificate usage in development environments

### DISCUSSION THREAD

**Discussion by jschne1der** [International-Data-Spaces-Association/IDS-Connector-Framework#29] [Ideas]
Currently I'm developing a connector with the framework as a basis. 
To follow best practices, I'm trying to remove all (dev) certs from our repository. For local development [mkcert](https://github.com/FiloSottile/mkcert) is used to generate all certs and CA. This will repeated on every dev machine. 

The problem arises with collecting a DAT: because there is no integrated mock of the DAPS or similar, every connector cert has to be registered with one of the current active DAPS (e.g. <https://daps.aisec.fraunhofer.de>). This is somewhat complicated and I think local development should be possible without external server dependencies. 

I understand why there is no direct possibility to deactivate the DAT call.   

As a workaround I'm trying to configure https://github.com/Fraunhofer-AISEC/omejdn-server with my local docker and mkcert setup. 

**tl;dr:** The Framework is missing a way to be run completely local for dev purposes.

**Comment by ghost** [comment]
Thanks for the feedback, I will take it to the next internal telco and ask for feedback there.
